cvat/apps/dataset_manager/tests/test_formats.py

```diff
@@ -484,6 +576,498 @@ def test_frames_outside_are_not_generated(self):
+                    "Datumaro 1.0", # not uploaded
+                    "Segmentation mask 1.1", # changed points values
+                    "WiderFace 1.0", # issue #XXX
+                ]:
+                    continue
+
+                print("dump_format_name:", dump_format_name)
+
+                # create task
+                images = self._generate_task_images(3)
+                task = self._create_task(tasks["main"], images)
+
+                # create annotations
+                if dump_format_name in ["MOT 1.1", "MOTS PNG 1.0", "PASCAL VOC 1.1", "Segmentation mask 1.1", "TFRecord 1.0", "YOLO 1.1", "ImageNet 1.0", "WiderFace 1.0", "VGGFace2 1.0"]:
+                    self._create_annotations(task, dump_format_name, "dafault")
+                else:
+                    self._create_annotations(task, dump_format_name, "random")
+
+                task_id = task["id"]
+                task_ann = TaskAnnotation(task_id)
+                task_ann.init_from_db()
+                task_data = TaskData(task_ann.ir_data, Task.objects.get(pk=task_id))
+                extractor = CvatTaskDataExtractor(task_data)
+                data_from_task_before_upload = datumaro.components.project.Dataset.from_extractors(extractor)
+
+                # dump annotations
+                url = f"/api/v1/tasks/{task_id}/annotations"
+                data = {
+                    "format": dump_format_name,
+                    "action": "download",
+                }
+                file_zip_name = osp.join(osp.dirname(__file__), 'assets', f'{test_name}_{dump_format_name}.zip')
+                self._download_file(url, data, self.admin, file_zip_name)
+                self._check_downloaded_file(file_zip_name)
+
+                # remove annotations
+                response = self._delete_request(url, self.admin)
+                self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+
+                # upload annotations
+                if dump_format_name in ["CVAT for images 1.1", "CVAT for video 1.1"]:
+                    upload_format_name = "CVAT 1.1"
+                else:
+                    upload_format_name = dump_format_name
+
+                url = f"/api/v1/tasks/{task_id}/annotations?format={upload_format_name}"
+                with open(file_zip_name, 'rb') as binary_file:
+                    response = self._upload_file(url, binary_file, self.admin)
+                    self.assertEqual(response.status_code, status.HTTP_202_ACCEPTED)
+
+                # equals annotations
+                task_ann = TaskAnnotation(task_id)
+                task_ann.init_from_db()
+                task_data = TaskData(task_ann.ir_data, Task.objects.get(pk=task_id))
+                extractor = CvatTaskDataExtractor(task_data)
+                data_from_task_after_upload = datumaro.components.project.Dataset.from_extractors(extractor)
+                compare_datasets(self, data_from_task_before_upload, data_from_task_after_upload)
```

Comment:
Consider using `require_images=True` along with `save_images=True` in line 648 for dataset export to compare image data. This can affect the kind of exported format in several formats.

Reply:
Done. Added it actions.